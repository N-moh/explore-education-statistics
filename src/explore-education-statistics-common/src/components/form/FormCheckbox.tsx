import classNames from 'classnames';
import React, { ChangeEventHandler } from 'react';
import styles from './FormCheckbox.module.scss';

export type CheckboxChangeEventHandler = ChangeEventHandler<HTMLInputElement>;

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: CheckboxChangeEventHandler;
  small?: boolean;
  value: string | string[] | number;
}

const FormCheckbox = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  small = false,
  value,
}: Props) => {
  return (
    <div
      className={classNames('govuk-checkboxes__item', {
        [styles.itemSmall]: small,
      })}
    >
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
