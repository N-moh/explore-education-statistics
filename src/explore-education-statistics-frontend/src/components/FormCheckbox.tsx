import React, { ChangeEventHandler, FunctionComponent } from 'react';

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  value: string | string[] | number;
}

const FormCheckbox: FunctionComponent<Props> = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  value,
}) => {
  return (
    <div className="govuk-checkboxes__item">
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
