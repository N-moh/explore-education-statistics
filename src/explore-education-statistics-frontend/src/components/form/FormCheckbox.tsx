import React, { ChangeEventHandler, FunctionComponent } from 'react';

export type CheckboxChangeEventHandler<
  T = HTMLInputElement
> = ChangeEventHandler<T>;

interface Props {
  checked?: boolean;
  id: string;
  hint?: string;
  label: string;
  name: string;
  onChange?: CheckboxChangeEventHandler;
  value: string | string[] | number;
}

export const FormCheckbox: FunctionComponent<Props> = ({
  checked,
  id,
  hint,
  label,
  name,
  onChange,
  value,
}) => {
  return (
    <div className="govuk-checkboxes__item">
      <input
        aria-describedby={hint ? `${id}-item-hint` : undefined}
        className="govuk-checkboxes__input"
        checked={checked}
        id={id}
        name={name}
        onChange={onChange}
        type="checkbox"
        value={value}
      />
      <label className="govuk-label govuk-checkboxes__label" htmlFor={id}>
        {label}
      </label>
      {hint && (
        <span
          id={`${id}-item-hint`}
          className="govuk-hint govuk-checkboxes__hint"
        >
          {hint}
        </span>
      )}
    </div>
  );
};

export default FormCheckbox;
