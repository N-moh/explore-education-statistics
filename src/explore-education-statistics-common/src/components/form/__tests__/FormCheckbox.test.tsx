import React from 'react';
import { render } from 'react-testing-library';
import FormCheckbox from '../FormCheckbox';

describe('FormCheckbox', () => {
  test('renders correctly with required props', () => {
    const { container } = render(
      <FormCheckbox
        name="test"
        id="test-checkbox"
        label="Test checkbox"
        value="true"
      />,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('renders correctly when `hint` is provided', () => {
    const { container, getByLabelText, getByText } = render(
      <FormCheckbox
        name="test"
        hint="Click me to proceed"
        id="test-checkbox"
        label="Test checkbox"
        value="true"
      />,
    );

    expect(getByText('Click me to proceed')).toHaveAttribute(
      'id',
      'test-checkbox-item-hint',
    );
    expect(getByLabelText('Test checkbox')).toHaveAttribute(
      'aria-describedby',
      'test-checkbox-item-hint',
    );

    expect(container.innerHTML).toMatchSnapshot();
  });
});
