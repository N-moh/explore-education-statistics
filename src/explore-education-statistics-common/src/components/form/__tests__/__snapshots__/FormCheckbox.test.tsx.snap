// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCheckbox renders correctly when \`hint\` is provided 1`] = `

<div class="govuk-checkboxes__item">
  <input aria-describedby="test-checkbox-item-hint"
         class="govuk-checkboxes__input"
         id="test-checkbox"
         name="test"
         type="checkbox"
         value="true"
  >
  <label class="govuk-label govuk-checkboxes__label"
         for="test-checkbox"
  >
    Test checkbox
  </label>
  <span id="test-checkbox-item-hint"
        class="govuk-hint govuk-checkboxes__hint"
  >
    Click me to proceed
  </span>
</div>

`;

exports[`FormCheckbox renders correctly with required props 1`] = `

<div class="govuk-checkboxes__item">
  <input class="govuk-checkboxes__input"
         id="test-checkbox"
         name="test"
         type="checkbox"
         value="true"
  >
  <label class="govuk-label govuk-checkboxes__label"
         for="test-checkbox"
  >
    Test checkbox
  </label>
</div>

`;
